src/CVRF/cvrf_eval.c

```diff
@@ -0,0 +1,550 @@
+
+	struct oscap_iterator *branches = cvrf_product_tree_get_branches(tree);
+	while (oscap_iterator_has_more(branches)) {
+		branch = oscap_iterator_next(branches);
+
+		if (!strcmp(cvrf_branch_get_branch_type(branch), "Product Version")) {
+			full_name = cvrf_branch_get_cvrf_product_name(branch);
+
+			if (oscap_str_endswith(product_id, cvrf_product_name_get_product_id(full_name))) {
+				rpm_name = cvrf_product_name_get_cpe(full_name);
+				break;
+			}
+		}
+
+	}
+	oscap_iterator_free(branches);
+	return rpm_name;
+}
+
+struct cvrf_rpm_attributes *parse_rpm_name_into_components(struct cvrf_model_eval *eval, const char *product_id) {
+
+	struct cvrf_rpm_attributes *attributes = cvrf_rpm_attributes_new();
+	const char *rpm_name = get_rpm_name_from_cvrf_product_id(eval, product_id);
+	attributes->full_package_name = strdup(rpm_name);
+
+	struct oscap_string *rpm_name_suffix = oscap_string_new();
+	struct oscap_string *rpm_release = oscap_string_new();
+	struct oscap_string *evr_format = oscap_string_new();
+	int dashes_size = 1;
+	int periods_size = 1;
+
+	char **split_by_dashes = oscap_split(strdup(rpm_name), "-");
+	while (split_by_dashes[dashes_size-1] != NULL)
+		dashes_size++;
+	for (int index = 0; index < dashes_size-3; index++) {
+		oscap_string_append_string(rpm_name_suffix, split_by_dashes[index]);
+		if (index < dashes_size-4)
+			oscap_string_append_char(rpm_name_suffix, '-');
+	}
+	attributes->rpm_name = strdup(oscap_string_get_cstr(rpm_name_suffix));
+	attributes->rpm_version = strdup(split_by_dashes[dashes_size-3]);
+
+	char **split_by_periods = oscap_split(split_by_dashes[dashes_size-2], ".");
+	while (split_by_periods[periods_size-1] != NULL)
+		periods_size++;
+	for (int index = 0; index < periods_size-3; index++) {
+		oscap_string_append_string(rpm_release, split_by_periods[index]);
+		if (index < periods_size-4)
+			oscap_string_append_char(rpm_release, '.');
+	}
+	attributes->rpm_release = strdup(oscap_string_get_cstr(rpm_release));
+	attributes->rpm_architecture = strdup(split_by_periods[periods_size-3]);
+
+	char *product_id_dup = strdup(product_id);
+	char **split_by_colon = oscap_split(product_id_dup, ":");
+	char *name_with_epoch = oscap_strdup(split_by_colon[1]);
+	int name_length = strlen(attributes->rpm_name);
+
+	if (name_length == strlen(name_with_epoch)) {
+		oscap_string_append_char(evr_format, '0');
+	} else {
+		for (int epoch_index = name_length+1; epoch_index < strlen(name_with_epoch); epoch_index++)
+			oscap_string_append_char(evr_format, name_with_epoch[epoch_index]);
+	}
+	oscap_string_append_char(evr_format, ':');
+	oscap_string_append_string(evr_format, attributes->rpm_version);
+	oscap_string_append_char(evr_format, '-');
+	oscap_string_append_string(evr_format, attributes->rpm_release);
+
+	attributes->evr_format = strdup(oscap_string_get_cstr(evr_format));
+
+	oscap_string_free(evr_format);
+	oscap_string_free(rpm_name_suffix);
+	oscap_string_free(rpm_release);
+	return attributes;
```

Comment:
we discussed this in person. this function will most likely have to be reworked or even removed.